src/ngraph_encapsulate_op.cc

```diff
@@ -715,6 +639,106 @@ class NGraphEncapsulateOp : public OpKernel {
+    std::vector<const Tensor*> static_input_map;
+    std::shared_ptr<ngraph::Function> ng_function;
+    std::shared_ptr<ngraph::runtime::Executable> ng_exec;
+    ng::runtime::Backend* op_backend;
+
+    // Get ngraph executable and inputs information
+    OP_REQUIRES_OK(ctx, GetNgExec(ctx, ng_exec, input_shapes, static_input_map,
+                                  op_backend));
+
+    NGRAPH_VLOG(4)
+        << "NGraphEncapsulateOp::Compute got ngraph executable for cluster "
+        << m_ngraph_cluster;
+
+    int time_func_create_or_lookup = function_lookup_or_create.ElapsedInMS();
+    event_func_maybe_create.Stop();
+
+    NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute got graph for cluster "
+                   << m_ngraph_cluster;
+
+    Timer create_or_lookup_tensors;
+
+    if (m_freshness_tracker == nullptr) {
+      auto creator = [](NGraphFreshnessTracker** tracker) {
+        *tracker = new NGraphFreshnessTracker();
+        return Status::OK();
+      };
+      OP_REQUIRES_OK(
+          ctx, ctx->resource_manager()->LookupOrCreate<NGraphFreshnessTracker>(
+                   ctx->resource_manager()->default_container(),
+                   "ngraph_freshness_tracker", &m_freshness_tracker, creator));
+    }
+
+    NGRAPH_VLOG(4)
+        << "NGraphEncapsulateOp::Compute got freshness tracker for cluster "
+        << m_ngraph_cluster;
+
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
+    bool log_copies = false;
+    OP_REQUIRES_OK(ctx, IsCopyLogEnabled(m_graph_id, log_copies));
+    copy_log_str << "KERNEL[" << type_string() << "]: " << name()
+                 << " ,GraphID " << m_graph_id << "\n";
+#endif
+
+    // Allocate tensors for input arguments.
+    ngraph::Event event_alloc_input("Input: maybe create", name(), "");
+
+    vector<shared_ptr<ng::runtime::Tensor>> ng_inputs;
+    std::vector<std::unique_ptr<ngraph::Event>> input_copy_events;
+    int ng_input_tensor_size_in_bytes = 0;
+
+    OP_REQUIRES_OK(ctx,
+                   AllocateTensorInput(ctx, ng_exec, input_shapes, op_backend,
+                                       ng_inputs, input_copy_events));
+
+    NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute allocated argument tensors "
+                      "for cluster "
+                   << m_ngraph_cluster;
+
+    // Now write the events back
+    for (auto& next : input_copy_events) {
+      ngraph::Event::write_trace(*next.get());
+    }
+    event_alloc_input.Stop();
+
+    vector<shared_ptr<ng::runtime::Tensor>> ng_outputs;
+    int ng_output_tensor_size_in_bytes = 0;
+
+    // Allocate tensors for the output results.
+    ngraph::Event event_alloc_output("Output: maybe create", name(), "");
+    std::vector<std::pair<void*, std::shared_ptr<ng::runtime::Tensor>>>
+        output_caches;
+    OP_REQUIRES_OK(ctx,
+                   AllocateTensorOutput(ctx, ng_exec, input_shapes, op_backend,
+                                        ng_inputs, ng_outputs, output_caches));
```

Comment:
Not sure why we need to pass the output_caches. The function signature for AllocateTensorInputs and AllocateTensorOutputs should be same/similar, i would think

Reply:
@shresthamalik returning output caches because it is used here https://github.com/tensorflow/ngraph-bridge/blob/master/src/ngraph_encapsulate_op.cc#L774